### PR TITLE
[test-only][full-ci] added test to create odt file inside deleted parent folder

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
Added test to create odt file inside a deleted parent folder.

## Related Issue

- https://github.com/owncloud/ocis/issues/9682

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 1]
```diff
@@ -119,4 +119,24 @@ public function userCreatesFileInsideFolderInSpaceUsingWopiEndpoint(string $user
...
+	public function userTriesToCreateAFileInsideDeletedFolderUsingWopiEndpoint(string $user, string $file): void {
+		$parent_container_id = $this->featureContext->getStoredFileID();
+		$this->featureContext->setResponse(
+			HttpRequestHelper::post(
+				$this->featureContext->getBaseUrl() . "/app/new?parent_container_id=$parent_container_id&filename=$file",
+				$this->featureContext->getStepLineRef(),
+				$user,
+				$this->featureContext->getPasswordForUser($user)
+			)
+		);
```
Comment (amrita-shrestha) [inline tests/acceptance/bootstrap/CollaborationContext.php:140]: DRY concept

**Comment by PrajwolAmatya** [comment]
Lets wait for the PR https://github.com/owncloud/ocis/pull/9978 to get merged first.

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 2]
```diff
@@ -105,15 +106,18 @@ public function userChecksTheInformationOfFileOfSpaceUsingOffice(string $user, s
...
 		$this->featureContext->setResponse(
-			HttpRequestHelper::post(
-				$this->featureContext->getBaseUrl() . "/app/new?parent_container_id=$parent_container_id&filename=$file",
+			CollaborationHelper::createFile(
+				$this->featureContext->getBaseUrl(),
 				$this->featureContext->getStepLineRef(),
 				$user,
-				$this->featureContext->getPasswordForUser($user)
+				$this->featureContext->getPasswordForUser($user),
+				$parent_container_id,
```
Comment (amrita-shrestha) [inline tests/acceptance/bootstrap/CollaborationContext.php:119]:
```suggestion
				$parent_container_id,
```

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 3]
```diff
@@ -105,15 +106,18 @@ public function userChecksTheInformationOfFileOfSpaceUsingOffice(string $user, s
 	 * @param string $space
 	 *
 	 * @return void
+	 * @throws GuzzleException
 	 */
 	public function userCreatesFileInsideFolderInSpaceUsingWopiEndpoint(string $user, string $file, string $folder, string $space): void {
 		$parent_container_id = $this->spacesContext->getResourceId($user, $space, $folder);
```
Comment (amrita-shrestha) [inline tests/acceptance/bootstrap/CollaborationContext.php:112]:
```suggestion
		$parentContainerId = $this->spacesContext->getResourceId($user, $space, $folder);
```

**Comment by nabim777** [comment]
LGTM :+1:

**Pull request merged**